src/draw_request.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <memory>
+
+#include <SDL2/SDL.h>
+
+enum GraphicPriority {
+    OVERLAY,
+    UI,
+    FRONT,
+    MIDDLE,
+    BACK,
+    BACKGROUND
+};
+
+class DrawRequest {
+
+public:
+    explicit DrawRequest(SDL_Texture *texture,
+                SDL_Rect texture_rect,
+                GraphicPriority z_level)
+    :texture_(texture)
+    ,texture_rect_(texture_rect)
+    ,z_level_(z_level)
+    {}
+
+    SDL_Texture* texture() const {
+        return texture_;
+    }
+    
+    const SDL_Rect& texture_rect() const {
+        return texture_rect_;
+    }
+    
+    GraphicPriority z_level() const {
+        return z_level_;
+    }
+
+
+private:
+    SDL_Texture* texture_;
+    SDL_Rect texture_rect_;
+    GraphicPriority z_level_;
```

Comment:
Is there a point to store the z_level_ inside GraphicPriority?

The `void GraphicsHandler::draw(SDL_Texture *texture, int x_pos, int y_pos, GraphicPriority priority)` function has `draw_queue_.at(priority).push_back(DrawRequest(texture, texture_rect, priority));`

which stores the DrawRequest in a vector according to it's priority already right?

I'm mostly wondering if you think we will ever have to query for the z_level_ in the future.